lib/gen_socket_client.ex

```diff
@@ -321,16 +319,24 @@ defmodule Phoenix.Channels.GenSocketClient do
   end
 
   defp reinit(state) do
-    :ets.delete_all_objects(state.message_refs)
+    :erlang.erase()
     if (state.transport_mref != nil), do: Process.demonitor(state.transport_mref, [:flush])
     %{state | transport_pid: nil, transport_mref: nil}
   end
 
   defp transport(state),
-    do: Map.take(state, [:transport_mod, :transport_pid, :message_refs, :serializer])
-
-  defp next_ref(topic, message_refs),
-    do: :ets.update_counter(message_refs, topic, 1, {topic, 0})
+    do: Map.take(state, [:transport_mod, :transport_pid, :serializer])
+
+  defp next_ref(topic) do
+    case :erlang.get(topic) do
```

Comment:
Also, I think we should tag the procdict key, so it doesn't get mixed up with something by the user. For example, `{__MODULE__, topic}`, or something like that.